src/network/network_content_gui.cpp

```diff
@@ -810,7 +810,9 @@ class NetworkContentListWindow : public Window, ContentCallback {
 			case WID_NCL_NAME:
 				if (this->content.SortType() == widget - WID_NCL_CHECKBOX) {
 					this->content.ToggleSortOrder();
-					if (this->content.size() > 0) this->list_pos = (int)this->content.size() - this->list_pos - 1;
+					if (this->content.size() > 0) {
+						this->list_pos = (this->list_pos - 1 > this->content.size()) ? 0 : this->content.size() - this->list_pos - 1;
```

Comment:
shouldn't this be `+ 1`, as in:

`this->list_pos = (this->list_pos + 1 > this->content.size()) ? 0 : this->content.size() - this->list_pos - 1;`

?